service/api_task.go

```diff
@@ -229,7 +229,7 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 
 	// mark task as finished
 	deactivatePrevious := projectRef.ShouldDeactivatePrevious()
-	if projectParser.ShouldDeactivatePrevious() {
+	if projectParser != nil && projectParser.ShouldDeactivatePrevious() {
```

Comment:
I'm not sure we should ever be getting here (although this is still a reasonable check). I'm thinking what's actually happening is that we're erroring on 208 but not returning.  Should we be returning? It looks like we _are_ hitting errors but I would think that we shouldn't be. We may want to add a grip log to ParserProjectFindOneVersion to detail if we're using the passed in version or looking one up, and what the version ID is.
https://mongodb.splunkcloud.com/en-US/app/search/search?q=search%20index%3Devergreen%20metadata.level%3E%3D70%20500&display.page.search.mode=verbose&dispatch.sample_ratio=1&earliest=1632331860&latest=1632333686&workload_pool=standard_perf&sid=1632333699.411703